src/main.rs

```diff
@@ -79,36 +79,28 @@ fn main() -> Result<()> {
     // Would be nice if we had a way to parse this. The first flags we see will be going to nushell
     // then it'll be the script name
     // then the args to the script
-    let mut collect_arg_nushell = false;
-    for arg in std::env::args().skip(1) {
+    let mut args = std::env::args().skip(1);
+    while let Some(arg) = args.next() {
         if !script_name.is_empty() {
-            args_to_script.push(if arg.contains(' ') {
-                format!("`{}`", arg)
-            } else {
-                arg
-            });
-        } else if collect_arg_nushell {
-            args_to_nushell.push(if arg.contains(' ') {
-                format!("`{}`", arg)
-            } else {
-                arg
-            });
-            collect_arg_nushell = false;
+            args_to_script.push(escape_quote_string(&arg));
         } else if arg.starts_with('-') {
             // Cool, it's a flag
-            if arg == "-c"
-                || arg == "--commands"
-                || arg == "--testbin"
-                || arg == "--log-level"
-                || arg == "--config"
-                || arg == "--env-config"
-                || arg == "--threads"
-                || arg == "-t"
-            {
-                collect_arg_nushell = true;
-            }
+            let flag_value = match arg.as_ref() {
+                "--commands" | "-c" => {
+                    // FIXME: Use proper quoting. `escape_quote_string()` can't be used for now due to https://github.com/nushell/nushell/issues/5383.
+
+                    args.next().map(|a| format!("`{}`", a))
```

Comment:
Using proper quoting will fix https://github.com/nushell/nushell/issues/5175

Reply:
Feel free to propose a 'proper quoting' fix too. :)